atomic_reactor/plugins/pre_hide_files.py

```diff
@@ -0,0 +1,122 @@
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import absolute_import
+
+import os
+
+from atomic_reactor.plugin import PreBuildPlugin
+from atomic_reactor.plugins.pre_reactor_config import get_hide_files
+from atomic_reactor.util import df_parser
+from atomic_reactor.constants import INSPECT_CONFIG
+
+
+class HideFilesPlugin(PreBuildPlugin):
+    key = 'hide_files'
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow):
+        """
+        Plugin initializer
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        """
+        super(HideFilesPlugin, self).__init__(tasker, workflow)
+        self.start_lines = []
+        self.end_lines = []
+
+    def run(self):
+        """
+        run the plugin
+        """
+        try:
+            self.hide_files = get_hide_files(self.workflow)
+        except KeyError:
+            self.log.info("Skipping hide files: no files to hide")
+            return
+
+        if not self.hide_files['tmpdir'] or not self.hide_files['files']:
+            self.log.info("tmpdir or files for hide_files are empty")
+            return
+
+        self._get_start_file_lines()
+        self._get_end_file_lines()
+
+        for parent_image_id in self.workflow.builder.parent_images:
+            self._update_dockerfile(parent_image_id)
+
+    def _get_start_file_lines(self):
+        for file_to_hide in self.hide_files['files']:
+            self.start_lines.append('RUN mv -f {} {} || :'.format(file_to_hide,
+                                                                  self.hide_files['tmpdir']))
+
+    def _get_end_file_lines(self):
+        for file_to_hide in self.hide_files['files']:
+            file_base = os.path.basename(file_to_hide)
+            tmp_dest = os.path.join(self.hide_files['tmpdir'], file_base)
+
+            self.end_lines.append('RUN mv -fZ {} {} || :'.format(tmp_dest, file_to_hide))
+
+    def _update_dockerfile(self, parent_image_id):
+        dfp = df_parser(self.workflow.builder.df_path)
+        add_start_lines = []
+        add_end_lines = []
+
+        inspect = self.workflow.builder.parent_image_inspect(parent_image_id)
+        inherited_user = inspect.get(INSPECT_CONFIG).get('User', '')
+
+        parent_structure = self._find_parent_structure(dfp, parent_image_id)
+        last_stage_structure = self._find_last_stage_structure(dfp, parent_structure)
+
+        if inherited_user:
+            add_start_lines.append('USER 0')
+
+        add_start_lines.extend(self.start_lines)
+
+        if inherited_user:
+            add_start_lines.append('USER {}'.format(inherited_user))
+
+        dfp.add_lines_at(parent_structure, *add_start_lines, after=True)
+
+        final_user_line = "USER " + inherited_user if inherited_user else None
+        # Look for the last USER after the last FROM... by looking in reverse
+        for insndesc in reversed(dfp.structure):
```

Comment:
This needs to be scoped for the a given stage. Instead of the whole dfp.structure, you need to only look at the lines relevant to this stage.

Reply:
yeah I see now